src/main/kotlin/org/elm/ide/intentions/exposure/ExposeIntention.kt

```diff
@@ -19,37 +20,40 @@ open class ExposeIntention : ExposureIntentionBase<ExposeIntention.Context>() {
     override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
         val exposingList = getExposingList(element) ?: return null
 
-        // check if the caret is on the identifier that names the exposable declaration
         val decl = element.parent as? ElmExposableTag ?: return null
         if (decl.nameIdentifier != element) return null
 
         return when {
-            decl is ElmUnionVariant -> {
-                // might be nice to support this in the future (making a union type fully exposed)
-                null
-            }
-
-            decl is ElmFunctionDeclarationLeft && !decl.isTopLevel ->
-                null
-
-            !exposingList.exposes(decl) ->
-                createContext(decl, exposingList)
-
-            else ->
-                null
+            // might be nice to support this in the future (making a union type fully exposed)
+            decl is ElmUnionVariant -> null
+            decl is ElmFunctionDeclarationLeft && !decl.isTopLevel -> null
+            !exposingList.exposes(decl) -> createContext(decl, exposingList)
+            else -> null
         }
     }
 
-    /**
-     * Creates a [Context] based on the passed in parameters. Overriding subclasses can return null if they find that
-     * the passed in [decl] isn't valid for their particular intention.
-     */
     protected open fun createContext(decl: ElmExposableTag, exposingList: ElmExposingList): Context? =
         Context(decl.name, exposingList)
 
     override fun invoke(project: Project, editor: Editor, context: Context) {
-        WriteCommandAction.writeCommandAction(project).run<Throwable> {
+        // The actual mutation (no write wrappers inside this lambda)
```

Comment:
What does “no write wrappers inside this lambda” mean?

Reply:
It means that nothing actually gets written inside this function -- it applies the edits to the text, so that a preview can be generated, but it doesn't actually write the file.